pkg/rpcmiddleware/validator/validator.go

```diff
@@ -31,7 +31,7 @@ type validatorWithContext interface {
 }
 
 type validator interface {
-	Validate() error
+	ValidateFields() error
```

Comment:
1. The signature is `ValidateFields(...string) error`. You're not matching anything here.
2. The interface name is also no longer appropriate with `ValidateFields`
3. Also look at the contextual version